File: android/src/main/kotlin/com/compress/all/flutter_compress/ImageConfig.kt
package com.compress.all.flutter_compress

/** Parsed mirror of the Dart [ImageCompressConfig]. */
data class ImageConfig(
    val format: String,
    val quality: Int,
    val targetSizeKB: Int?,
    val maxWidth: Int?,
    val maxHeight: Int?,
    val keepExif: Boolean,
) {
    companion object {
        fun fromMap(m: Map<String, Any?>) = ImageConfig(
            format = m["format"] as? String ?: "jpeg",
            quality = (m["quality"] as? Number)?.toInt() ?: 85,
            targetSizeKB = (m["targetSizeKB"] as? Number)?.toInt(),
            maxWidth = (m["maxWidth"] as? Number)?.toInt(),
            maxHeight = (m["maxHeight"] as? Number)?.toInt(),
            keepExif = m["keepExif"] as? Boolean ?: false,
        )
    }
}

File: ios/flutter_compress/Sources/flutter_compress/ImageConfig.swift
import Foundation

/// Parsed mirror of the Dart `ImageCompressConfig`.
struct ImageConfig {
  let format: String
  let quality: Int
  let targetSizeKB: Int?
  let maxWidth: Int?
  let maxHeight: Int?
  let keepExif: Bool

  init(map: [String: Any]) {
    format = map["format"] as? String ?? "jpeg"
    quality = (map["quality"] as? NSNumber)?.intValue ?? 85
    targetSizeKB = (map["targetSizeKB"] as? NSNumber)?.intValue
    maxWidth = (map["maxWidth"] as? NSNumber)?.intValue
    maxHeight = (map["maxHeight"] as? NSNumber)?.intValue
    keepExif = map["keepExif"] as? Bool ?? false
  }
}
